contrib/k8s/examples/mysql/mysql-dbms-only-instance.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceInstance
+metadata:
+  name: my-mysqldb-instance
```

Comment:
How come sometimes we are using `my-mysql-database-*` and sometimes `my-mysqldb-*`? If we are trying to make it clear that one is a server and another is a db instance maybe use `mysql-server-*` and `mysql-database-*` instead? Or maybe `dbms` since that's another term you are using elsewhere.

Reply:
Yeah, that was an omission! Thanks for catching it, I've updated it to my-mysql-dbms-instance to follow the pattern.

Reply:
🎵 _Carolyn just found a bug! 💃  Carolyn just found a bug! 🕺_ 🎵

Reply:
Ya-- this looks off.

Opinion incoming...

We should name example _resources_ like follows:

1. Prefix all resources with `my-` or `example-` or _something_ to make the name feel a little more "customized." We don't want to fool anyone (mostly people who do not know k8s as well as we do) into thinking that resource names have to match service or plan names. (I was favoring `my-`, before, but now feel `example-` might be more appropriate.
2. Manifest file names should match _resource_ names; not service names.
3. If the instance provisioned by a given manifest is a DBMS, it should say so in the resource name and file name. Likewise, if the thing provisioned is a database, that should be denoted clearly as well. (I think we should avoid "db" because in both writing and speech it's too easily to mis-say/mis-hear or mis-write/mis-read "db" when "dbms" might have been intended.
4. If the instance being provisioned is the "all-in-one" DBMS + single database, I think rather than confuse people with "all-in-one," we should just refer to such instances as "example-mysql-instance"-- not denoting DBMS _or_ database. Since this is unspecific, let's treat this like the default case. Specifying that you want a DBMS or database is advanced usage.
5. Make resource and file names for bindings include the name of the instance they bind to. `example-mysql-database-binding` binds to `example-mysql-instance`.

Bottom line is that we just need naming to be crystal clear. Feel free to push back on these recommendations if you think I am nuts.

Reply:
Maybe also if we treat "all-in-one" as the base case and everything else as "advanced usage," perhaps for each set of examples, we can move the "advanced" manifests off into an `avanced/` subdir.